src/entry.js

```diff
@@ -69,8 +71,7 @@ class Entry {
       clock: entry.clock,
     })
 
-    const pubKey = await keystore.importPublicKey(entry.key)
-    await keystore.verify(entry.sig, pubKey, Buffer.from(JSON.stringify(e)))
+    return identity.provider.verify(entry.sig, entry.key, encode(e))
```

Comment:
We could simplify this a little bit more and instead of passing `identity`, we pass only the `IdentityProvider` (which is what this function uses as opposed to needing the full details of the Identity).